s3po/backends/s3.py

```diff
@@ -93,3 +93,9 @@ def func():
             multi.upload_part_from_file(StringIO(data), count)
             multi.complete_upload()
             return True
+
+    def list(self, bucket, prefix=None, delimiter=None, headers=None):
```

Comment:
Maybe it's only an issue for the swift backend since it seems to fully materialize the list before creating the generator. I think enforcing a limit is wise maybe even with a default of something like 10 or even 1 to avoid ballooning memory.